### ISSUE TITLE
Clicking a story in the storysource resets the addon panel

### ISSUE THREAD

**Issue by backbone87** [storybookjs/storybook#5891]
When you use the storysource addon and click a story inside it, you switch to that story, but the addon panel resets and you see the initial tab insteadof keeping the storysource addon panel open at the position it was.

**Comment by medmunds** [comment]
@shilman I don't think this is specific to addon-storysource. I'm seeing exactly the same behavior in 5.0.1 with addon-knobs and addon-actions (but not loading storysource). And switching stories with keyboard shortcuts will also reset to the first tab.

I think the bug may actually be that if you have more than one addon panel registered, any navigation between stories switches the addon panel back to the first tab. In 4.x, navigating between stories preserved the currently-active tab in the addon panel.

<details>
<summary>My addons.js</summary>
```javascript
import '@storybook/addon-knobs/register';
import '@storybook/addon-actions/register';
import '@storybook/addon-options/register';
import '@storybook/addon-viewport/register';
```
</details>

**Comment by backbone87** [comment]
imo for best UX, the last actively clicked tab should be marked, so when switching to a story that does not contain the tab and then back to the previous story, this tab would be opened again (because on the other story, no active tab selection was made)

**Comment by shilman** [comment]
Related https://github.com/storybooks/storybook/issues/5857

**Comment by Hypnosphi** [comment]
@medmunds As far is I understand, this issue is about navigation throw storysource addon panel, not through sidebar. What you're talking about sounds more like https://github.com/storybookjs/storybook/issues/6077

**Comment by medmunds** [comment]
@Hypnosphi yes, on re-reading, #6077 is the problem I encountered (navigating between stories in the side story list should maintain the active addon tab). This issue is similarly about maintaining the active addon tab, but when navigation between stories is performed within the storysource addon. (I'll update my earlier comment.)

**Comment by vanessayuenn** [comment]
Hi! I am going through and cleaning up old issues in the repo. We just shipped a ton of improvements in Storybook 7.0, and I suspect this bug might be fixed already. Can you please try it out? If this issue persists with the latest version, please open a new bug report with a 7.x reproduction. Thank you so much! 🙏🏼
